src/python/geoclaw/util.py

```diff
@@ -141,3 +150,121 @@ def inv_haversine(d,x1,y1,y2,Rsphere=Rearth,units='degrees'):
     if units=='degrees':
         dx = dx * RAD2DEG
     return dx
+
+
+def fetch_noaa_tide_data(station, begin_date, end_date, time_zone='GMT',
+                         datum='STND', units='metric', cache_dir='_cache'):
+    r"""Fetch water levels and tide predictions at given NOAA tide station.
+
+    The data is returned in 6 minute intervals between the specified begin and
+    end dates/times.  A complete specification of the NOAA CO-OPS API for Data
+    Retrieval used to fetch the data can be found at:
+
+        https://tidesandcurrents.noaa.gov/api/
+
+    By default, retrieved data is cached in the subdirectory '_cache' of the
```

Comment:
I am not sure we want to enable another way of caching local files as we already have a `scratch` directory in `$CLAW/geoclaw`.  I actually would not be opposed to using this instead except that sometimes multiple examples will use the same files which would mean redundant data download.  Given this I would probably suggest sticking these in the `scratch` directory instead.

Reply:
Updated to use scratch directory for caching by default.